### PR TITLE
Added Youtube support 

### PR DESCRIPTION
**Pull request by BastianKubaile**
I added Youtube to the supported Websites for this extension.

## Additions

I added Youtube to searchEngines.js. Thanks for the restructuring @Gazorby, it would have been way harder with out your additions. I also added Youtube to the Options Page and I found an issue doing it. Even though the page suggest it, there is no way to take away permissions from search engine websites. I will create a separate issue for this. 

Since Youtube uses endless scrolling, I had to change main.js quite a bit to support it. I used a MutationObserver that watches a container, specifiable in searchEngines.js, which basically reloads main.js when it detects any changes. There is some ugly cross referencing with this.known_results, which I want to change later.

### CODE REVIEW

File: src/manifest.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "permissions": ["storage"],
     "optional_permissions": [
         "http://*/*",
-        "https://*/*"
+        "https://*/*"
```
Comment (infokiller) [inline src/manifest.json]: Revert this change?

File: src/searchEngines.js [hunk 2]
```diff
@@ -205,6 +205,65 @@ const searchEngines = [
                     break;
             }
         }
+    },
+    {
+        canInit(){
+            return true;
+        }, 
+        urlPattern: /^(www|encrypted)\.youtube\./,
```
Comment (infokiller) [inline src/searchEngines.js]: There's no `encrypted.youtube` subdomain, so you can remove this part.

File: src/main.js [hunk 3]
```diff
@@ -339,14 +383,18 @@ const sleep = (milliseconds) => {
   return new Promise(resolve => setTimeout(resolve, milliseconds));
 }
 
-const loadOptions = extension.options.load();
+const startExt = () => {
```
Comment (infokiller) [inline src/main.js]: Please remame to `startExtension`, I like longer names :)

**Comment by infokiller** [comment]
Left a few minor comments, once you resolve them I will merge the PR

**Comment by BastianKubaile** [comment]
I resolved the issues you wanted.

**Comment by jinusean** [comment]
Does this support navigating in the home page and video page?
